Source/WebCore/dom/DOMStringList.h

```diff
@@ -64,9 +65,12 @@ class DOMStringList : public RefCounted<DOMStringList> {
     explicit DOMStringList(Vector<String>&& strings)
         : m_strings(WTFMove(strings))
     {
+        for (auto& string : m_strings)
+            m_stringsSet.add(string);
     }
 
     Vector<String> m_strings;
+    HashSet<String> m_stringsSet;
```

Comment:
This is a trade-off memory vs look-up speed. This will need to be A/B tested on Speedometer & memory benchmarks since this increases memory usage.

Reply:
Also, a HashSet + a Vector is basically a ListHashSet.

Reply:
I have just started perf A/B testing for this version. But yes, we should likely test a ListHashSet too.

Reply:
If you upload a version with a ListHashSet, I'll start A/B testing for that one too.

Reply:
Either is fine.

Reply:
But also, you claim to make a performance optimization but provided no numbers in the commit log. What is the motivation for this change? Which benchmark does this improve? Linear search in a vector is actually very fast (as fast or faster than a HashSet) when there are only a few items. In general I wouldn't expect a lot of items in the DOMTokenList. My point is that this change is controversial. I don't mind triggering A/B testing to see if this helps any of the benchmarks we are tracking (without regressing any) but I'm curious WHY you're proposing this change, what the motivation is.

Reply:
I guess you're right, `ListHashSet` maintains ordering but doesn't allow fast indexing. Anyway, Let's see what the perf results for your current PR show. If they progress one of the benchmarks we track, then we should do it. If not, I don't think it would be worth it.

Reply:
According to A/B testing, this PR does not progress the Speedometer benchmark. However, it is a ~0.7% memory regression on macOS (Membuster benchmark). I do not think this is viable.

Reply:
just for context. I filed the bug on bwo, just because of the comment in the code. 
And having real data to back it up is a good thing. Thanks @cdumez for the A/B testing. Maybe the comment should be modified or removed if it doesn't seem a good thing to do in the end.

Reply:
Sure, just drop the FIXME and use something like:
```
// Currently, all consumers of DOMStringList store fairly small lists and thus an O(n) algorithm is OK.
```